price-estimator/src/models.rs

```diff
@@ -79,11 +79,25 @@ pub struct MarketsResult {
     pub bids: Vec<TransitiveOrder>,
 }
 
+/// Sort orders by their price and merge orders with the same price into one.
+fn sanitize_orders(mut orders: Vec<TransitiveOrder>) -> Vec<TransitiveOrder> {
```

Comment:
nit: I think this should be part of the pricegraph API, specifically ask_prices and bid_prices.

Reply:
> I therefore think it could well be implemented in pricegraph directly (it would also not require sorting again).

pricegraph can be used by different projects. We shouldn't sort in pricegraph at all because different consumers might want different orderings. For example currently pricegraph orders both asks and bids by exchange by exchange_rate which is not the same thing as bid_prices. Someone else might want to order by volume etc.

Reply:
> We shouldn't sort in pricegraph at all because different consumers might want different orderings

I agree... I think pricegraph should just return the ask_prices and bid_prices in exchange rate order (which is how they are stored in the vec). My point was that I think this should be implemented as part of the ask_prices and bid_prices methods.

Reply:
The way i imagine this working out when implemented directly in `ask_prices` is more complicated if we don't want to collect into a vec internally. This is because we can't easily implement this with the iterator combinators so we we would have to create a custom struct implementing `DoubleEndedIterator`.
The need for merging the asks and bids is specific enough to the price estimator that I prefer to keep it there.

Reply:
> The need for merging the asks and bids is specific enough to the price estimator

I agree this is true for the orders, but not for `ask_prices` and `bid_prices`.